src/relations/backend_database.py

```diff
@@ -211,9 +211,13 @@ def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
         self.charm.unit.status = MaintenanceStatus("Initialising backend-database relation")
 
         try:
-            self.charm.check_pgb_running()
+            if not self.charm.check_pgb_running():
+                logger.debug("_on_database_created deferred: PGB not running")
+                event.defer()
+                return
```

Comment:
Check the output of `check_pgb_running` instead of relying on just an exception.